C#/en-us/Lesson02/AlienFinder-L2.cs

```diff
@@ -0,0 +1,125 @@
+using Neo.SmartContract.Framework;
+using Neo.SmartContract.Framework.Services.Neo;
+using System.Numerics;
+
+using Helper = Neo.SmartContract.Framework.Helper;
+
+public class AlienFinder : SmartContract
+{
+    // type name for the contract, 
+    const string AlienMapName = nameof(AlienFinder); 
+
+    public class Alien
+    {
+        private static BigInteger counter=0; 
+        
+        public uint Xna
+        { get; set; }
+        public string AlienName
+        { get; set; }
+        public uint BlockHeight
+        { get; private set; }
+        public BigInteger Id 
+        { get; private set; }
+
+        public Alien(uint xna, string alienName, uint blockHeight) 
+        {
+            Xna = xna; 
+            AlienName = alienName;
+            BlockHeight = blockHeight;
+
+            byte[] value = Storage.Get(Storage.CurrentContext, "counter"); 
+            if (value != null) { counter = value.ToBigInteger(); }
+            else { counter = 0; }
+
+            Id = counter++; 
+            Storage.Put(Storage.CurrentContext, "counter", counter); 
+        }
+    }
+
+    public static object Main(string operation, params object[] args) 
+    {
+        if (Runtime.Trigger == TriggerType.Verification) 
+        {
+            return false; 
+        }
+        switch (operation) {
+            case "generateAlien":
+                return GenerateAlien((string)args[0], (byte[])args[1]); 
+            case "query":
+                return Query((byte[]) args[0]); 
+            default:
+                return false; 
+        }
+    }
+
+    public static bool GenerateAlien(string alienName, byte[] owner) 
```

Comment:
@merlin217 If the `owner` here means the owner of the contract, then it should not be a parameter. Just store it through the contract constructor and check here inside.